src/aggregation/mod.rs

```diff
@@ -337,17 +338,79 @@ mod tests {
         )
     }
 
+    pub fn exec_request(agg_req: Aggregations, index: &Index) -> crate::Result<Value> {
+        exec_request_with_query(agg_req, index, None)
+    }
+    pub fn exec_request_with_query(
+        agg_req: Aggregations,
+        index: &Index,
+        query: Option<(&str, &str)>,
+    ) -> crate::Result<Value> {
+        let collector = AggregationCollector::from_aggs(agg_req);
+
+        let reader = index.reader()?;
+        let searcher = reader.searcher();
+        let agg_res = if let Some((field, term)) = query {
+            let text_field = reader.searcher().schema().get_field(field).unwrap();
+
+            let term_query = TermQuery::new(
+                Term::from_field_text(text_field, term),
+                IndexRecordOption::Basic,
+            );
+
+            searcher.search(&term_query, &collector)?
+        } else {
+            searcher.search(&AllQuery, &collector)?
+        };
+
+        let res: Value = serde_json::from_str(&serde_json::to_string(&agg_res)?)?;
```

Comment:
Can you add a commment explaining why we serialize and deserialize?
Wouldn't serde_json::to_value do the job ?

Reply:
yes, serde_json::to_value is better, I changed it

Reply:
ah no, I used that here to also test de/serialization roundtrip